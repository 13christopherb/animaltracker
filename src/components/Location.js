import React, {Component} from 'react';
import {connect} from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import _ from 'underscore';
import Animals from './Animals';


class Location extends Component {

    constructor(props) {
        super(props);
        this.state = {expanded: true}
    }

    render() {
        let animalsBySpecies = _.countBy(this.props.animals, (animal) => {
            return animal['species']
        });
        let speciesCount  = [];

        for (var property in animalsBySpecies) {
            if (animalsBySpecies.hasOwnProperty(property)) {
                speciesCount.push(property + '(' + animalsBySpecies[property] + ') ')
            }
        }

        return (
            <div>
                <a onTransitionEnd={this.onCollapseClick} className="btn btn-info btn-block" data-toggle="collapse"
                   href={'#'+this.props.location} role="button"
                   aria-expanded="false">
                    <div className='row'>
                    <div className='col-md-1 offset-md-5'>{this.props.location}</div>
                    <div className='col-md-1 '><small>{speciesCount}</small></div>
                    </div>
                </a>
                <div className="collapse show" aria-expanded="true" id={this.props.location}>
                    <Animals animals={this.props.animals}/>
                </div>
            </div>
        );
    }
}

function mapStateToProps({animals}, ownProps) {
    return {
        animals: ownProps.animals
    }
}

export default connect(
    mapStateToProps,
)(Location);