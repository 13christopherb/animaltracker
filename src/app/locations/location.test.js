import React from 'react';
import {shallow, mount} from 'enzyme';
import Location from './location';
import toJson from 'enzyme-to-json';

const animals = [{
    name: 'Nymeria',
    weight: 33,
    species: 'Dire Wolf',
    location: 'MBO'
},
    {
        name: 'Lady',
        weight: 33,
        species: 'Dire Wolf',
        location: 'MBO'
    }

];

describe('<Location />', () => {
    describe('render()', () => {
        test('should render the component', () => {
            const wrapper = shallow(<Location animals={animals} location="MBO"/>);

            expect(toJson(wrapper)).toMatchSnapshot()
        })
    })
});