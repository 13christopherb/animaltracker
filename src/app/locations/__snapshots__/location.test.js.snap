// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LocationContainer /> render() should render the component 1`] = `
<div>
  <a
    aria-expanded="false"
    className="btn btn-info btn-block"
    data-toggle="collapse"
    href="#undefined"
    role="button"
  >
    <div
      className="row"
    >
      <div
        className="col-md-1 offset-md-5"
      />
      <div
        className="col-md-1 "
      >
        <small />
      </div>
    </div>
  </a>
  <div
    aria-expanded="true"
    className="collapse show"
  >
    <Animals />
  </div>
</div>
`;
